From 008e7dc0ae46281360be62b7d69cd13e0a7f7e14 Mon Sep 17 00:00:00 2001
From: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Date: Sun, 7 Feb 2016 18:24:35 +0100
Subject: [PATCH] Link against libintl when needed

Since the kbd code uses gettext functionality, it should link with the
libintl library using the @INTLLIBS@ symbol provided by the gettext m4
macro. This is necessary on C libraries that don't provide a built-in
gettext functionality, and rely on the libintl library provided by
gettext, as is the case with the uClibc C library.

@INTLLIBS@ is empty when linking a separate library is not needed, as
is the case with the glibc C library.

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
---
 src/Makefile.am   | 2 +-
 tests/Makefile.am | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/Makefile.am b/src/Makefile.am
index 40e8083..d4e0583 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -80,7 +80,7 @@ mapscrn_CFLAGS = -DMAIN
 loadunimap_CFLAGS = -DMAIN
 
 noinst_LIBRARIES = libcommon.a libfont.a
-LDADD = libcommon.a libfont.a libkeymap/libkeymap.la
+LDADD = libcommon.a libfont.a libkeymap/libkeymap.la @INTLLIBS@
 
 install-exec-hook:
 	for i in psfaddtable psfgettable psfstriptable; do \
diff --git a/tests/Makefile.am b/tests/Makefile.am
index 1976333..848fd31 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -4,7 +4,7 @@ AM_CPPFLAGS = \
 	-DDATADIR=\"$(srcdir)\" -DBUILDDIR=\"$(builddir)\"
 
 AM_CFLAGS = $(CHECK_CFLAGS)
-LDADD  = $(top_builddir)/src/libkeymap/libkeymap.la $(CHECK_LIBS)
+LDADD  = $(top_builddir)/src/libkeymap/libkeymap.la $(CHECK_LIBS) @INTLLIBS@
 
 EXTRA_DIST = \
 	alt-is-meta.in \
-- 
2.6.4

